sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml

```diff
@@ -15,21 +15,32 @@ provider:
       patternNot: 
         - "(\\w)\\1{5,}"
         - "(?i)test|abcd|1234"
-        - "(?i)\
-        YOURTENANTNAME|\
-        mycompany|\
-        yourorg|\
-        team[_-]?name|\
-        example|\
-        contoso|\
-        acmecorp"
+        - "(?i)YOURTENANTNAME|mycompany|yourorg|(s|ex)ample"
+        - "(?i)team[_-]?name"
+        - "contoso|acmecorp"
 
   rules:
     - rspecKey: S6721
       id: ms-teams-webhook-urls
       metadata:
         name: Microsoft Teams Webhook Urls
+      detection:
+        matching:
+          # Structure:
+          # https://tenant-name.webhook.office.com/webhookb2/guid@guid/IncomingWebhook/connectorId/guid
+          # An MS GUID is in the following format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX where X is a hex digit
+            pattern: "((?:https://)?\
+                    [a-z0-9_-]{1,50}\\.webhook\\.office\\.com/webhookb2/\
+                    [a-z0-9\\-]{1,50}@[a-z0-9\\-]{1,50}/IncomingWebhook/[a-z0-9]{1,50}/[a-z0-9\\-]{1,50})"
       examples:
+        - text: |
+            # Noncompliant code example
```

Comment:
The noncompliant code example should raise an issue. Maybe the example needs to be adjusted.

Reply:
done